base/pkg/entry.jl

```diff
@@ -197,7 +197,11 @@ function clone(url::AbstractString, pkg::AbstractString)
         end
     catch err
         isdir(pkg) && Base.rm(pkg, recursive=true)
-        rethrow(err)
+        if err.msg == "Unexpected HTTP status code: 404"
```

Comment:
> Are you suggesting updating line 200 to something like
> if isa(err, Base.LibGit2.Error.GitError) && err.msg == "Unexpected HTTP status code: 404"? If so, do all GitError exceptions have a message field?

Yes. `GitError` is a concrete type, so all exceptions of this type have this field.

> How do you feel about
> if sprint(show, err) == "GitError(Code:ERROR, Class:Net, Unexpected HTTP status code: 404)"?

That's not what I meant. My suggestion was that we could add another field to `GitError` which would store the HTTP error code. But that's not really required for this PR.